src/qt/bitcoin.cpp

```diff
@@ -282,7 +282,7 @@ void BitcoinApplication::parameterSetup()
 }
 
 void BitcoinApplication::SetPrune(bool prune, bool force) {
-     optionsModel->SetPrune(prune, force);
+    optionsModel->SetPruneTargetGB(prune ? DEFAULT_PRUNE_TARGET_GB : 0, force);
 }
```

Comment:
@ryanofsky 

Agree with all your suggestions.

As this PR has 3 ACKs already, is it acceptable to implement your suggestions in a following PR, e.g., #17453?

Reply:
> As this PR has 3 ACKs already, is it acceptable to implement your suggestions in a following PR, e.g., #17453?

Of course yes, my suggestions are always just suggestions, and I wouldn't ACK a PR that I didn't think was ready to merge. I'd slightly prefer cleanup here or in a standalone PR instead of #17453 to make #17453 more focused and understandable, but it doesn't matter too much.

Reply:
Done in the latest push.